src/sql/workbench/contrib/welcome/gettingStarted/browser/initialSetupWizard.ts

```diff
@@ -0,0 +1,452 @@
+import { Registry } from 'vs/platform/registry/common/platform';
+import { IWorkbenchActionRegistry, Extensions } from 'vs/workbench/common/actions';
+import { SyncActionDescriptor } from 'vs/platform/actions/common/actions';
+import { KeyCode } from 'vs/base/common/keyCodes';
+import { Button } from 'sql/base/browser/ui/button/button';
+import { ConfigurationTarget, IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { onUnexpectedError } from 'vs/base/common/errors';
+import { IStorageService, StorageScope } from 'vs/platform/storage/common/storage';
+import { buttonSecondary, buttonSecondaryBorder, buttonSecondaryHoverColor, contentSeperatorLine, setupModalShadowOne, setupModalShadowTwo, setupOverlay } from 'sql/platform/theme/common/colorRegistry';
+const intialSetupWizardKey = 'workbench.initialSetup';
+
+const $ = dom.$;
+interface InitialSetupWizardData {
+	key: string;
+	order: string;
+	header: string;
+	subHeader: string;
+	body: string;
+	docs?: string;
+	elmClass: string;
+	id: string;
+	btnId: string;
+	btnText: string;
+	subjectImage?: string;
+	popupImage: string;
+	interaction?: string;
+	interactionLabel?: string;
+	name?: string;
+}
+
+const initialSetupWizardData: InitialSetupWizardData[] = [
+	{
+		key: 'preview_features',
+		order: '1',
+		header: localize('initialSetupWizard.setupHeader', "Azure Data Studio Setup"),
+		subHeader: localize('initialSetupWizard.subheaderPreviewFeatures', "Opt in to use preview features"),
+		body: localize('initialSetupWizard.previewFeaturesBody', "Azure Data Studio releases continuously. To access new features and user experiences as soon as they’re released, we recommend that you opt in. "),
+		docs: 'https://aka.ms/ads-preview-features',
+		elmClass: 'ads-initial-setup-wizard-preview-features',
+		id: 'ads-initial-setup-wizard-preview-features',
+		btnId: 'ads-initial-setup-wizard-preview-features-btn',
+		btnText: localize('initialSetupWizard.done', "Done"),
+		popupImage: './../../gettingStarted/media/enablePreviewFeatures.svg',
+		interaction: 'checkbox',
+		interactionLabel: 'Opt in (recommended)',
+		name: 'previewFeatures'
+	}
+];
+
+const IS_OVERLAY_VISIBLE = new RawContextKey<boolean>('interfaceOverviewVisible', false);
+let gettingStartedSetupWizard: GettingStartedSetupWizard;
+
+export class GettingStartedSetupWizardAction extends Action {
+	public static readonly ID = 'workbench.action.createInitialSetupWizard';
+	public static readonly LABEL = localize('initialSetupWizard', "Initial Setup Wizard");
+
+	constructor(
+		id: string,
+		label: string,
+		@IInstantiationService private readonly instantiationService: IInstantiationService
+	) {
+		super(id, label);
+	}
+
+	public run(): Promise<void> {
+		if (!gettingStartedSetupWizard) {
+			gettingStartedSetupWizard = this.instantiationService.createInstance(GettingStartedSetupWizard);
+		}
+		gettingStartedSetupWizard.create();
+		return Promise.resolve();
```

Comment:
Note - for cases like this you can just make the functions async and then you can just return; (or do nothing since it's the end of the function)